tests/test_uploader.py

```diff
@@ -40,112 +40,83 @@
-
-
-ckan_uploader = Uploader(config['auth_token'],
+ckan_uploader = Uploader(config['base_url'],
+                config['auth_token'],
                 config['organization_id'],
                 config['dataset_id'],
-                config['api']
             )
 
-file = FileSystem('./test/fixtures/sample.csv')
+file = FileSystem('./tests/sample_file/dailyprices.csv')
 
 def test_can_instantiate_uploader():
-    datahub = Uploader(config['auth_token'],
+    datahub = Uploader(config['base_url'],
+                config['auth_token'],
                 config['organization_id'],
                 config['dataset_id'],
-                config['api']
             )
-    assert datahub.api == config['api']
+    assert datahub.base_url == config['base_url']
+
+
+get_jwt_from_ckan_authz_json = {
+                    "help": "http://localhost:5000/api/3/action/help_show?name=authz_authorize",
+                    "success": True,
+                    "result": {
+                      "requested_scopes": [
+                          "obj:myorg/dataset-name/*:write"
+                      ],
+                      "granted_scopes": [],
+                      "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzY29wZXMiOiIiLCJ==",
+                      "user_id": "admin",
+                      "expires_at": "2020-04-22T20:08:41.102934+00:00"
+                    }
+                }
+
+request_file_upload_actions_json = {
+            'transfer': 'basic',
+            'objects': [
+                {
+                    'oid': '8857053d874453bbe8e7613b09874e2d8fc9ddffd2130a579ca918301c31b369',
+                    'size': 123,
+                    'authenticated': True,
+                    'actions': {
+                    'upload': {
+                        'href': 'https://myaccount.blob.core.windows.net/mycontainer/my-blob',
+                        'header': access_granter_config['headers'],
+                        'expires_in': 86400,
+                    },
+                    'verify': {
+                        'href': 'https://some-verify-callback.com',
+                        'header': {
+                        'Authorization': 'Bearer TOKEN',
+                        },
+                        'expires_in': 86400,
+                    },
+                    },
+                  },
+              ],
+          }
+
+@mock.patch("lib.util.ckan_auth_api.get_jwt_from_ckan_authz", return_value=get_jwt_from_ckan_authz_json)
+@mock.patch("lib.util.ckan_auth_api.request_file_upload_actions", return_value=request_file_upload_actions_json)
+@mock.patch("lib.util.ckan_upload_api.upload_to_storage", return_value=True)
+@mock.patch("lib.util.ckan_upload_api.verify_upload", return_value=True)
+def test_push_works_with_packaged_dataset(get_jwt_from_ckan_authz_mock, request_file_upload_actions_mock,
+                                          upload_to_storage_mock, verify_upload_mock):
+    import pdb; pdb.set_trace()
```

Comment:
Committing code with pdb debugging in it is unusual ...

Reply:
Sorry about this(pdb) and pyc files!